rust/src/datafusion/logical_expr.rs

```diff
@@ -112,6 +112,22 @@ pub fn resolve_expr(expr: &Expr, schema: &Schema) -> Result<Expr> {
                         right: right.clone(),
                     }));
                 }
+                (Expr::Column(l), Expr::BinaryExpr(r)) => {
```

Comment:
What case is this covering? `y = (x < 2)`?

Reply:
This is covering cases like 
```x = 2 + 2```
```x = 1 * -1```
```x = 2 - 2 + -1```
as well as 
```y = x - 2```
The idea was to cover essentially any expression of literals on right hand side **with valid type on left **. I am wondering if ```y = x < 2``` is a valid filter ?
The expression can be parsed alright but I am getting an error in the planning.
```
ValueError: LanceError(IO): Error during planning: Float64 = Boolean can't be evaluated because there isn't a common type to coerce the types to
```
This is mostly because, x < 2 for each value in x will evaluate to boolean which ofc can't be coerced to type of x which is float64.

Reply:
> I am wondering if y = x < 2 is a valid filter ?

No, it's probably not. I wasn't sure what the use of it was, so I was confused.

Could you add a comment in line explaining what cases it is trying to cover? Want to make sure people reading this code later know what it does :)